scripts/zones/Rabao/npcs/Leodarion.lua

```diff
@@ -73,15 +75,14 @@ function onEventFinish(player,csid,option)
         player:delKeyItem(tpz.ki.OLD_TRICK_BOX);
         player:setCharVar("trueWillCS",2);
     elseif (csid == 99) then
-        if (player:getFreeSlotsCount() < 1) then
-            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,13782);
-        else
-            player:delKeyItem(tpz.ki.LARGE_TRICK_BOX);
-            player:addItem(13782);
-            player:messageSpecial(ID.text.ITEM_OBTAINED,13782); -- Ninja Chainmail
-            player:setCharVar("trueWillCS",0);
-            player:addFame(NORG,30);
-            player:completeQuest(OUTLANDS,tpz.quest.id.outlands.TRUE_WILL);
+        if npcUtil.completeQuest(player, OUTLANDS, tpz.quest.id.outlands.TRUE_WILL, {
+                item = 13782, -- Ninja Chainmail
+                fameArea = NORG,               
+                title = tpz.title.PARAGON_OF_NINJA_EXCELLENCE,
+                var = "trueWillCS"
+            })
+        then
+            player:delKeyItem(tpz.ki.LARGE_TRICK_BOX)
```

Comment:
I think it's possible to specify 2 key items in completeQuest(), is there a technical reason this should happen only after?

Reply:
Sorry, I misunderstood this is taking an item whereas completeQuest is giving one.